tiledb/sm/query/readers/reader_base.h

```diff
@@ -161,6 +161,13 @@ class ReaderBase : public StrategyBase {
   /** The query condition. */
   QueryCondition& condition_;
```

Comment:
(general comment) I know this is pre-existing code, but seems like it would be clearer if this was held in a `std::optional` instead of using `condition_.empty()`.